python/taulabs/uavtalk.py

```diff
@@ -57,29 +57,46 @@ def process_stream(uavo_defs, use_walltime=False, gcs_timestamps=False):
 
     received = 0
 
-    packet_bytes = ''
+    buf = ''
+    buf_offset = 0
+
+    pending_pieces = []
 
     while True:
+        # If we don't have data buffered, join up any chunks we've been given
+        # to ensure pending_pieces is empty for the rest of this loop.
+        #
+        # in other words, don't mix the pending_pieces drain model and the
+        # buffer concatenation model within a loop iteration.
+        #
+        # 10k chosen here to be bigger than any plausible uavo
```

Comment:
Using a large constant here is fine, but you could also compute it (once) from the uavo_defs list that was passed in as the generator was being started.